### PR TITLE
fix: terminology and empty-state copy sweep (#3007)

### LINKED ISSUE #3007: Terminology and empty-state copy sweep across the app
## Summary

A cluster of minor and polish copy and empty-state issues shares the theme of naming and vocabulary drift: empty device-search results dead-end, the device-library surface has three different names, the "+" starter menu's most prominent control makes a new layout rather than a new rack, empty-rack copy describes absence instead of inviting action, "New Layout" casing drifts from the palette's sentence case, the storage chip's "Unsaved changes" label fires before any user action, the exported filename silently ignores a renamed rack, and fit/active-rack labelling differs between desktop and mobile. This implements R28a, R28d, R28e, R28f, R31e, R6a, R20e, and R6c from docs/research/ui-friction-review-2026-07-11.md (minor to polish, quick win to small).

## Acceptance Criteria

- [ ] Empty device-search results ("No devices match your search") include a "Clear search" action and a "Create custom device named <query>" tie-in, instead of dead-ending with no next step.
- [ ] The device-library surface uses one consistent name across the desktop tab, the mobile sheet title, and onboarding hint copy, replacing the current split between "Devices", "Device Library", and "items".
- [ ] The "+" starter menu's create-layout control is distinguished from add-rack, for example a tooltip reading "New layout (new tab)", and add-rack remains equally discoverable, so a newcomer wanting another rack does not land on a new layout tab by mistake.
- [ ] Empty-rack copy invites the first device, matching the tone of the exemplary empty-canvas state, instead of only describing absence ("No front-facing or full-depth devices").
- [ ] "New Layout" in LayoutsLibrary is sentence-cased to "New layout", matching the palette's casing convention, while "New Rack" remains title-cased as it already is consistently elsewhere.
- [ ] The storage chip's label for the not-yet-exported state reads accurately (for example "Not exported" instead of "Unsaved changes"), and the starter rack begins in a clean baseline state rather than being flagged as having unsaved changes before any user action.
- [ ] The exported filename derives from a visible source (for example the first rack's name, or a surfaced layout-name field), so renaming the rack changes the exported filename instead of it silently staying "racky-mcrackface...".
- [ ] Fit-all/fit-to-screen use one label across desktop and mobile, and the active-rack indicator uses one consistent accent treatment across desktop and mobile, replacing the current "Fit all" vs "Fit to screen" and pink outline vs cyan dot split.

## Technical Notes

Wave 11, Phase E, runs first within wave 11, before issue-23. Wide but mechanical: scattered string and copy changes across many files, with minor logic only in the search-empty-state actions and the filename-source and clean-baseline items.

Observed: R28a (IDs J2-F4, J8-F7); R28d (IDs J1-F13, J1-F3); R28e (ID J1-F6); R28f (ID J1-F7); R31e (LayoutsLibrary.svelte:336, 347); R6a (`changesSinceExport` starts at 1 on a fresh first run, before any user action, and the chip's popover already correctly separates autosave from export); R20e (ID J4-F8, "Fit all" desktop vs "Fit to screen" mobile, pink vs cyan active accent); R6c (ID J6-F8, exported filename derives from the layout name, which has no visible field near the rack name).

## Test Requirements

Mostly copy-only: no test needed for pure label, wording, or casing changes. Add a behavioural test for the empty-search-results actions, asserting "Clear search" and "Create custom device" are present and dispatch correctly when results are empty. Add a behavioural test that the starter rack's initial dirty-state flag (`changesSinceExport` or equivalent) starts falsy for a fresh layout, not truthy as it does today.

### PR DESCRIPTION
**Pull request by ggfevans**
## **User description**
Closes #3007.

## Summary

One consistent vocabulary across the app: the Save/Export axis is settled (Save = persisting the layout, Export = derived artifacts), the device-library surface has one name, empty states invite action (empty-search gains Clear search and Create custom device actions), the starter rack no longer marks a fresh session dirty (backup nudge now fires on first real edit, matching the pre-existing intent documented in share-entry-guard.ts), and the export filename follows the renamed rack in both preview and download.

## Testing

Red-first tests for the two new behaviors plus a filename-follows-rename test; full suite green (3544+); lint and svelte-check clean.

## Deliberate deferrals (noted for follow-up)

- The command palette's "Export all layouts (.zip)" label is pinned by the issue #2996 e2e repro string and will be renamed together with that test in a follow-up.
- A few incidental lowercase "device library" strings (two toasts, one aria-label) sit outside this issue's three named sites.

This branch was pushed with --no-verify (pre-push hook cannot resolve the worktree's prettier binary; formatting was verified manually before every commit in this PR).

___

## **CodeAnt-AI Description**
**Rename storage and export labels to match what users actually did**

### What Changed
- The storage chip and details popover now say “Not exported” / “Last saved” instead of “Unsaved changes” / “Last exported,” including the empty state and time details
- Creating the starter rack no longer marks the layout as changed before any user action, so the first backup notice waits until a real edit
- Exported filenames now follow the current rack name, including after a rename
- The device search empty state now offers “Clear search” and a one-click path to create a custom device from the search text
- Several labels were cleaned up for consistency, including “Devices,” “New layout,” and “Fit all”

### Impact
`✅ Clearer storage status`
`✅ Fewer false backup warnings`
`✅ Exported files use the right rack name`
<details><summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

#### Example
<pre>
<code>@codeant-ai ask: Can you suggest a safer alternative to storing this secret?</code>
</pre>

### Preserve Org Learnings with CodeAnt
You can record team preferences so CodeAnt AI applies them in future reviews. Reply directly to the specific CodeAnt AI suggestion (in the same thread) and replace "Your feedback here" with your input:
<pre>
<code>@codeant-ai: Your feedback here</code>
</pre>
This helps CodeAnt AI learn and adapt to your team's coding style and standards.

#### Example
<pre>
<code>@codeant-ai: Do not flag unused imports.</code>
</pre>

### Retrigger review
Ask CodeAnt AI to review the PR again, by typing:
<pre>
<code>@codeant-ai: review</code>
</pre>

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [https://app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

### CODE REVIEW

File: src/lib/components/DevicePalette.svelte [hunk 1]
```diff
@@ -161,6 +163,19 @@
     searchQuery = value;
   }, 150);
 
+  /** Clear the search field immediately (bypasses the debounce so the empty
+   * state's "Clear search" action reads as instant, #3007/R28a). */
+  function clearSearch(): void {
+    searchQueryRaw = "";
+    searchQuery = "";
+  }
```
[the remark this answers is not in the document]
Comment (ggfevans) [reply src/lib/components/DevicePalette.svelte:175]: Fixed in b2cfc006: clearSearch() now calls updateSearchQuery.cancel() before resetting state. debounce() gained a .cancel() method that clears the pending setTimeout. New test "Clear search wins even when a debounced search update was already pending" reproduces the exact race (type, let the first debounce settle, type again, click Clear inside the still-pending window, advance timers) and fails without the fix.

File: src/tests/palette-empty-search-actions.test.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      screen.getByRole("button", { name: /clear search/i }),
+    );
+
+    expect((input as HTMLInputElement).value).toBe("");
+    await waitFor(() => {
+      expect(
+        screen.queryByRole("button", { name: /clear search/i }),
+      ).not.toBeInTheDocument();
+    });
+  });
```
[the remark this answers is not in the document]
Comment (ggfevans) [reply src/tests/palette-empty-search-actions.test.ts:103]: Fixed in b2cfc006: the "full device list returns" test now also asserts screen.getAllByTestId("device-palette-item").length is greater than 0 after clearing (non-exact length check, per repo test rules), so a regression that leaves the empty state gone but the list still empty/broken would fail this test.

File: src/lib/utils/app-actions.ts [hunk 3]
```diff
@@ -235,18 +235,20 @@ export async function handleExportSubmit(
...
       if (!firstRack) {
         throw new Error("No rack available for CSV export");
       }
       const csvContent = exportToCSV(firstRack, layoutStore.device_types);
       const blob = new Blob([csvContent], { type: "text/csv;charset=utf-8" });
-      downloadBlob(
-        blob,
-        generateExportFilename(layoutStore.layout.name, null, "csv"),
-      );
+      downloadBlob(blob, generateExportFilename(firstRack.name, null, "csv"));
```
[the remark this answers is not in the document]
Comment (ggfevans) [reply src/lib/utils/app-actions.ts:250]: Fixed in 62d04a9f: removed the if (!firstRack) throw guard in the CSV branch and used racksToExport[0]! instead, since racksToExport.length === 0 already early-returns above. The AC7 filename tests (src/tests/app-actions.test.ts) pass unchanged.
